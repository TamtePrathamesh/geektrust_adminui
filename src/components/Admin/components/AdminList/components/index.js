export { default as AdminActions } from './AdminActions';
export { default as RenderAdminRow } from './RenderAdminRow';
