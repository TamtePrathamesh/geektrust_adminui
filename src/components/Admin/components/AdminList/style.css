@import '../../../../style/colors.css';

table {
  width: inherit;
}

input[type='checkbox'] {
  width: 15px;
  height: 15px;
}

.table thead tr:first-child {
  border-top: hidden;
}

.table thead th {
  border-bottom: 1px solid var(--grey_one);
}

.columnName {
  letter-spacing: 2px;
}