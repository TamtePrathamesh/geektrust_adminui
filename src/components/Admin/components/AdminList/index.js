import React from 'react';
import PropType from 'prop-types';
import { RenderAdminRow } from './components';
import { WithBoundary } from '../../../common';

import './style.css';

const AdminList = ({ adminList, filter, onDelete, onEdit, onSelectOrDeselectAll, onSelectOrDeselect }) => {
  const isAllSelected = adminList.every(({ selection }) => selection);

  return (
    <>
      <div className="table-responsive-sm">
        <table className="table">
          <thead>
            <tr>
              <th>
                <input
                  data-testid="selectAll"
                  type="checkbox"
                  checked={isAllSelected}
                  onChange={onSelectOrDeselectAll}
                />
              </th>
              <th scope="col" className="columnName">
                Name
              </th>
              <th scope="col" className="columnName">
                Email
              </th>
              <th scope="col" className="columnName">
                Role
              </th>
              <th scope="col" className="columnName">
                Actions
              </th>
            </tr>
          </thead>
          <tbody>
            {adminList &&
              adminList
                .filter(({ name, email, role }) => {
                  return (
                    name.toLowerCase().indexOf(filter.toLowerCase()) !== -1 ||
                    email.toLowerCase().indexOf(filter.toLowerCase()) !== -1 ||
                    role.toLowerCase().indexOf(filter.toLowerCase()) !== -1
                  );
                })
                .map(({ id, name, email, role, selection }) => (
                  <RenderAdminRow
                    key={id}
                    admin={{ id, name, email, role, selection }}
                    onDelete={onDelete}
                    onEdit={onEdit}
                    onSelectOrDeselect={onSelectOrDeselect}
                  />
                ))}
          </tbody>
        </table>
      </div>
    </>
  );
};

AdminList.propTypes = {
  filter: PropType.string,
  onDelete: PropType.func,
  onEdit: PropType.func,
  onSelectOrDeselectAll: PropType.func,
  onSelectOrDeselect: PropType.func,
  adminList: PropType.arrayOf(
    PropType.shape({
      id: PropType.string,
      name: PropType.string,
      email: PropType.string,
      role: PropType.string,
      selection: PropType.bool,
    })
  ),
};

export default WithBoundary(React.memo(AdminList));
