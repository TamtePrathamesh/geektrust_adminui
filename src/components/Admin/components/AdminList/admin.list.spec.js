import React from 'react';
import { screen, render } from '@testing-library/react';
import user from '@testing-library/user-event';
import AdminList from '.';

const admins = [
  { id: '1', name: 'xyz', email: 'an@g.c', role: 'member', selection: false },
  { id: '2', name: 'pqr', email: 'an@g.c', role: 'member', selection: false },
];

describe('AdminList', () => {
  it('should match snapshot', () => {
    const onEdit = jest.fn();
    const onDelete = jest.fn();
    const onSelectOrDeselectAll = jest.fn();
    const onSelectOrDeselect = jest.fn();
    expect(
      render(
        <AdminList
          adminList={admins}
          filter=""
          onDelete={onDelete}
          onEdit={onEdit}
          onSelectOrDeselect={onSelectOrDeselect}
          onSelectOrDeselectAll={onSelectOrDeselectAll}
        />
      ).container
    ).toMatchSnapshot();
  });

  it('should display only one row when filter is given as name', () => {
    const onEdit = jest.fn();
    const onDelete = jest.fn();
    const onSelectOrDeselectAll = jest.fn();
    const onSelectOrDeselect = jest.fn();
    expect(
      render(
        <AdminList
          adminList={admins}
          filter="xyz"
          onDelete={onDelete}
          onEdit={onEdit}
          onSelectOrDeselect={onSelectOrDeselect}
          onSelectOrDeselectAll={onSelectOrDeselectAll}
        />
      ).container
    ).toMatchSnapshot();
  });

  it('should select all row when selectAllRow checkbox clicked', () => {
    const onEdit = jest.fn();
    const onDelete = jest.fn();
    const onSelectOrDeselectAll = jest.fn(() => admins.map((i) => ({ ...i, selection: true })));
    const onSelectOrDeselect = jest.fn();

    render(
      <AdminList
        adminList={admins}
        filter=""
        onDelete={onDelete}
        onEdit={onEdit}
        onSelectOrDeselect={onSelectOrDeselect}
        onSelectOrDeselectAll={onSelectOrDeselectAll}
      />
    );
    expect(onSelectOrDeselectAll).toHaveBeenCalledTimes(0);
    user.click(screen.queryByTestId('selectAll'));
    expect(onSelectOrDeselectAll).toHaveBeenCalledTimes(1);
  });

  it('should call delete when button is clicked', () => {
    const onEdit = jest.fn();
    const onDelete = jest.fn((id) => admins.filter((i) => i.id !== id));
    const onSelectOrDeselectAll = jest.fn(() => admins.map((i) => ({ ...i, selection: true })));
    const onSelectOrDeselect = jest.fn();
    const { rerender } = render(
      <AdminList
        adminList={admins}
        filter=""
        onDelete={onDelete}
        onEdit={onEdit}
        onSelectOrDeselect={onSelectOrDeselect}
        onSelectOrDeselectAll={onSelectOrDeselectAll}
      />
    );

    expect(onDelete).toHaveBeenCalledTimes(0);
    user.click(screen.queryAllByTestId('deleteRow')[0]);
    expect(onDelete).toHaveBeenCalledTimes(1);
    rerender(
      <AdminList
        adminList={admins}
        filter=""
        onDelete={onDelete}
        onEdit={onEdit}
        onSelectOrDeselect={onSelectOrDeselect}
        onSelectOrDeselectAll={onSelectOrDeselectAll}
      />
    );
  });
});
