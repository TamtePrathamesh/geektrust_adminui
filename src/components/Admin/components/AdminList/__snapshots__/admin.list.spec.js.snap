// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminList should display only one row when filter is given as name 1`] = `
<div>
  <div
    class="table-responsive-sm"
  >
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            <input
              data-testid="selectAll"
              type="checkbox"
            />
          </th>
          <th
            class="columnName"
            scope="col"
          >
            Name
          </th>
          <th
            class="columnName"
            scope="col"
          >
            Email
          </th>
          <th
            class="columnName"
            scope="col"
          >
            Role
          </th>
          <th
            class="columnName"
            scope="col"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
          data-testid="tr"
        >
          <td>
            <input
              data-testid="selectRow"
              type="checkbox"
            />
          </td>
          <td
            class="columnName"
          >
            xyz
          </td>
          <td
            class="columnName"
          >
            an@g.c
          </td>
          <td
            class="columnName"
          >
            member
          </td>
          <td>
            <div
              class="actionItems"
            >
              <img
                alt="cant display"
                class="actionIcons"
                data-testid="editRow"
                src=""
              />
              <img
                alt="cant display"
                class="actionIcons"
                data-testid="deleteRow"
                src=""
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;

exports[`AdminList should match snapshot 1`] = `
<div>
  <div
    class="table-responsive-sm"
  >
    <table
      class="table"
    >
      <thead>
        <tr>
          <th>
            <input
              data-testid="selectAll"
              type="checkbox"
            />
          </th>
          <th
            class="columnName"
            scope="col"
          >
            Name
          </th>
          <th
            class="columnName"
            scope="col"
          >
            Email
          </th>
          <th
            class="columnName"
            scope="col"
          >
            Role
          </th>
          <th
            class="columnName"
            scope="col"
          >
            Actions
          </th>
        </tr>
      </thead>
      <tbody>
        <tr
          class=""
          data-testid="tr"
        >
          <td>
            <input
              data-testid="selectRow"
              type="checkbox"
            />
          </td>
          <td
            class="columnName"
          >
            xyz
          </td>
          <td
            class="columnName"
          >
            an@g.c
          </td>
          <td
            class="columnName"
          >
            member
          </td>
          <td>
            <div
              class="actionItems"
            >
              <img
                alt="cant display"
                class="actionIcons"
                data-testid="editRow"
                src=""
              />
              <img
                alt="cant display"
                class="actionIcons"
                data-testid="deleteRow"
                src=""
              />
            </div>
          </td>
        </tr>
        <tr
          class=""
          data-testid="tr"
        >
          <td>
            <input
              data-testid="selectRow"
              type="checkbox"
            />
          </td>
          <td
            class="columnName"
          >
            pqr
          </td>
          <td
            class="columnName"
          >
            an@g.c
          </td>
          <td
            class="columnName"
          >
            member
          </td>
          <td>
            <div
              class="actionItems"
            >
              <img
                alt="cant display"
                class="actionIcons"
                data-testid="editRow"
                src=""
              />
              <img
                alt="cant display"
                class="actionIcons"
                data-testid="deleteRow"
                src=""
              />
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
